public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { Checkbox, CollapsableSection, ColorValueEditor, Field, HorizontalGroup, Input } from '@grafana/ui';
+import { DashboardModel } from '../../state/DashboardModel';
+import { AnnotationQuery, DataSourceInstanceSettings } from '@grafana/data';
+import { DataSourcePicker } from 'app/core/components/Select/DataSourcePicker';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { useAsync } from 'react-use';
+import StandardAnnotationQueryEditor from 'app/features/annotations/components/StandardAnnotationQueryEditor';
+import { AngularEditorLoader } from './AngularEditorLoader';
+
+export const newAnnotation: AnnotationQuery = {
+  name: 'New annotation',
+  enable: true,
+  datasource: null,
+  iconColor: 'red',
+};
+
+type Props = {
+  editIdx: number | null;
+  dashboard: DashboardModel;
+};
+
+export const AnnotationSettingsEdit: React.FC<Props> = ({ editIdx, dashboard }) => {
+  const [annotation, setAnnotation] = useState(editIdx !== null ? dashboard.annotations.list[editIdx] : newAnnotation);
+
+  const { value: ds } = useAsync(async () => {
+    return await getDataSourceSrv().get(annotation.datasource);
+  }, [annotation.datasource]);
+
+  const onUpdate = (annotation: AnnotationQuery) => {
+    const list = [...dashboard.annotations.list];
```

Comment:
Not sure this copy is needed, the copy will be shallow anyway since the `list` contains objects.

Reply:
this is definitely required, and we do not need to do a deep clone here, just the item we are updating, 
The reason we need a new array is otherwise React components that take in the annotation array won't detect the change